src/builder/commands/ubuntu.rs

```diff
@@ -216,31 +222,18 @@ impl Distribution for Distro {
                         self.add_universe(ctx)?;
                     }
                 } else {
-                    info!("Unsupported distribution for eatmydata. Ingoring");
-                    self.do_eatmydata = false;
+                    info!("Unsupported distribution for eatmydata. Ignoring");
+                    self.eatmydata = EatMyData::No;
                 }
             }
             self.apt_update = false;
             apt_get_update::<&str>(ctx, &[])?;
-        }
-        if let Some(ref params) = eatmy {
-            apt_get_install(ctx, &[params.package])?;
-            match ctx.environ.entry("LD_PRELOAD".to_string()) {
-                Vacant(v) => {
-                    v.insert(params.preload.to_string());
-                }
-                Occupied(ref mut o) => {
-                    let mut m = o.get_mut();
-                    if m.len() > 0 {
-                        m.push(':');
-                        m.push_str(params.preload);
-                    } else {
-                        m.push_str(params.preload);
-                    }
-                }
+            if let Some(ref params) = eatmy {
+                apt_get_install(ctx, &[params.package], &EatMyData::No)?;
+                self.eatmydata = EatMyData::Preload(params.preload.to_string());
```

Comment:
Preload is currently a always is a static string, why do you clone it here, and then in each command invocation?